app/assets/javascripts/oxalis/model/binary/bucket_traversals.js

```diff
@@ -0,0 +1,172 @@
+  const uBucket = globalPositionToBucketPosition(startPosition, resolutions, zoomStep);
+  const lastBucket = globalPositionToBucketPosition(endPosition, resolutions, zoomStep);
+  // The integer variables X and Y are initialized to the starting voxel coordinates.
+  let [X, Y, Z] = uBucket;
+
+  const voxelSize = getBucketExtent(resolutions, zoomStep);
+
+  // In addition, the variables stepX and stepY are initialized to either 1 or -1 indicating whether X and Y are
+  // incremented or decremented as the ray crosses voxel boundaries (this is determined by the sign of the x and y components of → v).
+  const [stepX, stepY, stepZ] = v.map(el => Math.sign(el));
+  const step = [stepX, stepY, stepZ];
+  let [tMaxX, tMaxY, tMaxZ] = initializeTMax(u, v, [stepX, stepY, stepZ], voxelSize);
+
+  // Finally, we compute tDeltaX and tDeltaY. TDeltaX indicates how far along the ray we must move (in units of t) for the horizontal component
+  // of such a movement to equal the width of a voxel. Similarly, we store in tDeltaY the amount of movement along the ray which has a vertical component equal to the
+  // height of a voxel.
+  const [tDeltaX, tDeltaY, tDeltaZ] = [
+    Math.abs(voxelSize[0] / v[0]),
+    Math.abs(voxelSize[1] / v[1]),
+    Math.abs(voxelSize[2] / v[2]),
+  ];
+
+  const intersectedBuckets = [[X, Y, Z]];
+  let protection = 0;
+
+  const behindLastBucket = (dim, pos) => {
+    if (step[dim] < 0) {
+      return pos < lastBucket[dim];
+    } else if (step[dim] > 0) {
+      return pos > lastBucket[dim];
+    } else {
+      return false;
+    }
+  };
+
+  // Helper functions for visiting a next bucket
+  // Note: these functions mutate tMax[X|Y|Z] and intersectedBuckets
+  const visitNextX = () => {
+    const _X = X + stepX;
+    tMaxX = tMaxX + tDeltaX;
+    intersectedBuckets.push([_X, Y, Z]);
+    return _X;
+  };
+  const visitNextY = () => {
+    const _Y = Y + stepY;
+    tMaxY = tMaxY + tDeltaY;
+    intersectedBuckets.push([X, _Y, Z]);
+    return _Y;
+  };
+  const visitNextZ = () => {
+    const _Z = Z + stepZ;
+    tMaxZ = tMaxZ + tDeltaZ;
+    intersectedBuckets.push([X, Y, _Z]);
+    return _Z;
+  };
+
+  // Helper functions for visiting the next two buckets
+  const visitNextXY = () => {
+    const [_X, _Y] = [visitNextX(), visitNextY()];
+    intersectedBuckets.push([_X, _Y, Z]);
+    return [_X, _Y];
+  };
+  const visitNextXZ = () => {
+    const [_X, _Z] = [visitNextX(), visitNextZ()];
+    intersectedBuckets.push([_X, Y, _Z]);
+    return [_X, _Z];
+  };
+  const visitNextYZ = () => {
+    const [_Y, _Z] = [visitNextY(), visitNextZ()];
+    intersectedBuckets.push([X, _Y, _Z]);
+    return [_Y, _Z];
+  };
+
+  // Helper function for visiting the next three buckets
+  const visitNextXYZ = () => {
+    const [_X, _Y, _Z] = [visitNextX(), visitNextY(), visitNextZ()];
+    intersectedBuckets.push([_X, _Y, _Z]);
+    return [_X, _Y, _Z];
+  };
+
+  while (protection++ < 50000) {
```

Comment:
Since a small error in a while-loop can easily end in an endless loop, I added this loop protection to ensure that not more than 50 000 iterations are run. It's just a safe-guard which I would keep in the code. But I'll add a comment!